src/devices/LidarLiteV3/LidarLiteV3.cs

```diff
@@ -0,0 +1,358 @@
+                throw new ArgumentOutOfRangeException("Address must have 0-bit in the LSB.");
+            }
+
+            // Read in the unit's serial number.
+            Span<byte> rawData = stackalloc byte[2];
+
+            ReadBytes(Register.UNIT_ID, rawData);
+
+            // Write serial number to I2C_ID.
+            WriteRegister(Register.I2C_ID_HIGH, rawData[1]);
+            WriteRegister(Register.I2C_ID_LOW, rawData[0]);
+
+            // Write the new address.
+            WriteRegister(Register.I2C_SEC_ADDR, address);
+
+            // Disable the default address
+            WriteRegister(Register.I2C_CONFIG, 0x08);
+        }
+
+        #region Device Registers
+
+        /// <summary>
+        /// Get the last distance measurement in cm.
+        /// </summary>
+        public int LastDistance
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[2];
+                ReadBytes(Register.FULL_DELAY, rawData);
+                return BinaryPrimitives.ReadUInt16BigEndian(rawData);
+            }
+        }
+
+        /// <summary>
+        /// Get the difference between the current and last measurement resulting
+        /// in a signed (2's complement) 8-bit number in cm.
+        /// Positive is away from the device.
+        /// </summary>
+        public int DifferenceBetweenLastTwoDistances
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[1];
+                ReadBytes(Register.VELOCITY, rawData);
+                return (int)(sbyte)rawData[0];
+            }
+        }
+
+        /// <summary>
+        /// Get or set the various settings to control the acquistion behavior.
+        /// </summary>
+        public AcquistionSettings AcquistionSettings
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[1];
+                ReadBytes(Register.ACQ_CONFIG_REG, rawData);
+                return (AcquistionSettings)rawData[0];
+            }
+            set
+            {
+                WriteRegister(Register.ACQ_CONFIG_REG, (byte)value);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the maximum aquisition count limits the number of times
+        /// the device will integrate acquistions to find a correlation
+        /// record peak.
+        ///
+        /// Roughly correlates to: acq rate = 1/count and max
+        /// range = count^(1/4)
+        /// </summary>
+        private int MaximumAcquisitionCount
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[1];
+                ReadBytes(Register.SIG_COUNT_VAL, rawData);
+                return rawData[0];
+            }
+            set
+            {
+                WriteRegister(Register.SIG_COUNT_VAL, (byte)value);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the threshold of peak value that bypasses the internal algorithm.
```

Comment:
what internal algorithm will be bypassed? what will exactly happen when it's higher?